From 1c2c8b33fca0618da2a8aa78fd05a862ea8b5c3a Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Thu, 11 Oct 2018 18:36:52 +0530
Subject: [PATCH 0234/1051] net: octeontx2: Fix ucast NPC entry configuration
 sequence

Due to the changes made by patch
(92b8bf59e791 soc: octeontx2: Support to enable/disable default MCAM entries)
RSS action in NIXLF's ucast entry is getting overwritten while
adding MAC address to the entry. Hence moving RSS configuration
down i.e to after MAC address setting.

Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_pf.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
index f94205e33af3..db4081012771 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_pf.c
@@ -461,11 +461,6 @@ static int otx2_init_hw_resources(struct otx2_nic *pf)
 	if (err)
 		return err;
 
-	/* Initialize RSS */
-	err = otx2_rss_init(pf);
-	if (err)
-		return err;
-
 	for (lvl = 0; lvl < NIX_TXSCH_LVL_CNT; lvl++) {
 		err = otx2_txschq_config(pf, lvl);
 		if (err)
@@ -647,6 +642,11 @@ int otx2_open(struct net_device *netdev)
 	if (err)
 		goto err_disable_napi;
 
+	/* Initialize RSS */
+	err = otx2_rss_init(pf);
+	if (err)
+		goto err_disable_napi;
+
 	/* Register CQ IRQ handlers */
 	vec = pf->hw.nix_msixoff + NIX_LF_CINT_VEC_START;
 	for (qidx = 0; qidx < pf->hw.cint_cnt; qidx++) {
-- 
2.17.1

